shell/platform/linux/public/flutter_linux/fl_pixel_buffer_texture.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_PIXEL_BUFFER_TEXTURE_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_PIXEL_BUFFER_TEXTURE_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include "fl_texture.h"
+
+G_BEGIN_DECLS
+
+G_DECLARE_DERIVABLE_TYPE(FlPixelBufferTexture,
+                         fl_pixel_buffer_texture,
+                         FL,
+                         PIXEL_BUFFER_TEXTURE,
+                         FlTexture)
+
+struct _FlPixelBufferTextureClass {
+  FlTextureClass parent_class;
+
+  /**
+   * Virtual method called when Flutter wants to copy pixel buffer for
+   * populating this texture.
+   * @texture: an #FlPixelBufferTexture.
+   * @buffer: (out): pixel data.
+   * @format: (out): pixel buffer format (example GL_RGBA).
+   * @width: (inout): width of the texture in pixels.
+   * @height: (inout): height of the texture in pixels.
+   * @error: (allow-none): #GError location to store the error occurring, or
+   * %NULL to ignore.
+   *
+   * Returns: %TRUE on success.
+   */
```

Comment:
@robert-ancell What's the GLib convention for overridden methods? Usually we don't do that in our C++ code.

Reply:
@robert-ancell considers that extending class is better than static casting user data.

Reply:
Sorry, my comment was missing a key bit: that was supposed to say "the convention for **comments on** overridden methods".

Reply:
I had to look this one up. Virtual functions are actually documented in this style:
```c
/**
 * FlPixelBufferTextureClass:
 * @copy_pixels: ...
 */
```
See (GtkTextView](https://gitlab.gnome.org/GNOME/gtk/-/blob/d8c416e8b28bafc4d98079add3d34017247b559e/gtk/gtktextview.h#L141) for an example.

I think the current documentation is clearer to read though.